src/Message/PurchaseRequest.php

```diff
@@ -138,7 +1446,201 @@ public function getData()
+                    'cc'         => $this->getWorkPhoneCountryPrefix(),
+                    'subscriber' => $this->getWorkPhone(),
+                ])]
+            )
+            + ($this->get3DsRequestAuthenticationMethodData() !== null
+                || $this->get3DsRequestAuthenticationMethod() !== null
+                || $this->get3DsRequestAuthenticationTime() !== null
+                ? ['threeDSRequestorAuthenticationInfo' => array_filter([
+                    'threeDSReqAuthData'      => $this->get3DsRequestAuthenticationMethodData(),
+                    'threeDSReqAuthMethod'    => $this->get3DsRequestAuthenticationMethod(),
+                    'threeDSReqAuthTimestamp' => $this->formatDateTime(
+                        $this->get3DsRequestAuthenticationTime(),
+                        "YmdHi"
+                    ),
+                ])]
+                : []
+            )
+            + ($this->getCustomerAccountCreatedIndicator() !== null
+                || $this->getCustomerAccountCreatedDate() !== null
+                || $this->getCustomerAccountChangedIndicator() !== null
+                || $this->getCustomerAccountChangedDate() !== null
+                || $this->getCustomerPasswordChangedIndicator() !== null
+                || $this->getCustomerPasswordChangedDate() !== null
+                || $this->getCustomerPurchasesInLast6Months() !== null
+                || $this->getCustomerAccountCardProvisionsLast24Hours() !== null
+                || $this->getCustomerAccountTransactionsLast24Hours() !== null
+                || $this->getCustomerAccountTransactionsLastYear() !== null
+                || $this->getCustomerPaymentMethodCreatedIndicator() !== null
+                || $this->getCustomerPaymentMethodCreatedDate() !== null
+                || $this->getShippingAddressFirstUsedIndicator() !== null
+                || $this->getShippingAddressFirstUsedDate() !== null
+                || $this->getShippingNameCustomerNameMatch() !== null
+                || $this->getCustomerHasSuspiciousActivity() !== null
+                ? ['acctInfo' => array_filter([
+                    'chAccAgeInd'           => $this->getCustomerAccountCreatedIndicator(),
+                    'chAccDate'             => $this->formatDateTime($this->getCustomerAccountCreatedDate(), "Ymd"),
+                    'chAccChangeInd'        => $this->getCustomerAccountChangedIndicator(),
+                    'chAccChange'           => $this->formatDateTime(
+                        $this->getCustomerAccountChangedDate(),
+                        "Ymd"
+                    ),
+                    'chAccPwChangeInd'      => $this->getCustomerPasswordChangedIndicator(),
+                    'chAccPwChange'         => $this->formatDateTime(
+                        $this->getCustomerPasswordChangedDate(),
+                        "Ymd"
+                    ),
+                    'nbPurchaseAccount'     => $this->getCustomerPurchasesInLast6Months(),
```

Comment:
If purchases in last 6 months is `0` (or `'0'`), this will be removed by `array_filter()` (ditto other ones where 0 is a valid value).